src/XR.tsx

```diff
@@ -96,34 +92,106 @@ export function useHitTest(hitTestCallback: (hitMatrix: Matrix4, hit: XRHitTestR
   })
 }
 
-export function XR({ foveation = 0, children }: { foveation?: number; children: React.ReactNode }) {
-  const { gl, camera } = useThree()
+interface SessionStoreState {
+  set: SetState<SessionStoreState>
+  get: GetState<SessionStoreState>
+  session: XRSession | null
+}
+const sessionStore = create<SessionStoreState>((set, get) => ({ get, set, session: null }))
+
+export type XRCanvasEventType = 'sessionstart' | 'sessionend' | 'visibilitychange' | 'inputsourceschange'
+export class XRCanvasEvent extends Event {
+  readonly session: XRSession
+
+  constructor(type: XRCanvasEventType, session: XRSession) {
+    super(type)
+    this.session = session
+  }
+}
```

Comment:
This is a little annoying. Three doesn't emulate `XRSessionEvent` and `XREvents.tsx` exports an interface called `XREvent` which would more appropriately be `XRControllerEvent`.

I think I'll rename these in v4 so `XREvent` becomes more generic and extends to `XRControllerEvent`.